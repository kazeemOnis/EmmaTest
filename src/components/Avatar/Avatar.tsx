import React from 'react';
import styled from '../../theme';
import {colors} from '../../styles';

interface AvatarProps {
  image: string;
  imageWidth?: number;
  onPress?: () => void;
  active?: boolean;
}

const Wrapper = styled.TouchableOpacity<{active?: boolean; width: number}>`
  ${props => {
    return `
    height: ${props.width}px;
    width: ${props.width}px;
    border-radius: ${props.width / 2}px;
    justify-content: center;
    align-items: center;
    ${
      props.active &&
      `
      border-color: ${colors.aqua}; 
      border-width: 5px;`
    }
  `;
  }}
`;

const ImageWrapper = styled.View<{width: number}>`
  ${props => {
    return `
    height: ${props.width}px;
    width: ${props.width}px;
    border-radius: ${props.width / 2}px;
  `;
  }}
`;

const Image = styled.Image<{radius: number}>`
  width: 100%;
  height: 100%;
  border-radius: ${props => props.radius}px;
`;

export const Avatar = ({image, onPress, active, imageWidth}: AvatarProps) => {
  // Default width is 75px, maximum is 150px, minimum width is 50px;
  const width =
    imageWidth && imageWidth <= 150 && imageWidth >= 50 ? imageWidth : 75;

  return (
    <React.Fragment>
      {image && (
        <Wrapper
          testID="avatar"
          active={active}
          width={width + 5}
          onPress={() => {
            onPress && onPress();
          }}>
          <ImageWrapper width={width}>
            <Image source={{uri: image}} radius={width / 2} />
          </ImageWrapper>
        </Wrapper>
      )}
    </React.Fragment>
  );
};
