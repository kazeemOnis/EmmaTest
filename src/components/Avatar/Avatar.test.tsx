import React from 'react';
import {cleanup, fireEvent, render} from '@testing-library/react-native';
import {act} from 'react-test-renderer';
import {Avatar} from './Avatar';

afterEach(() => {
  cleanup();
});

const onPressMock = jest.fn();

const mockProps = {
  image: 'https://images.unsplash.com/photo-1579353977828-2a4eab540b9a',
  onPress: onPressMock,
};

const setup = (overrides = {}) => {
  const props = {...mockProps, ...overrides};
  const utils = render(<Avatar {...props} />);
  return {...utils, props};
};

describe('Avatar', () => {
  it('Should render avatar image', async () => {
    const {toJSON} = setup();
    await act(async () => {
      expect(toJSON()).toMatchSnapshot();
    });
  });

  it('Should click on the avatar and press action called', () => {
    const {getByTestId} = setup();
    fireEvent.press(getByTestId('avatar'));

    expect(onPressMock).toHaveBeenCalledTimes(1);
  });

  it('Should not render an avatar if no image is passed', async () => {
    const {queryByTestId, toJSON} = setup({
      image: null,
    });
    await act(async () => {
      expect(toJSON()).toMatchSnapshot();
      expect(queryByTestId('avatar')).toBeNull();
    });
  });
});
