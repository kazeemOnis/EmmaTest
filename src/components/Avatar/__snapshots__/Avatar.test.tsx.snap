// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Should not render an avatar if no image is passed 1`] = `null`;

exports[`Avatar Should render avatar image 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderRadius": 40,
      "height": 80,
      "justifyContent": "center",
      "opacity": 1,
      "width": 80,
    }
  }
  testID="avatar"
>
  <View
    style={
      [
        {
          "borderRadius": 37.5,
          "height": 75,
          "width": 75,
        },
      ]
    }
    width={75}
  >
    <Image
      radius={37.5}
      source={
        {
          "uri": "https://images.unsplash.com/photo-1579353977828-2a4eab540b9a",
        }
      }
      style={
        [
          {
            "borderRadius": 37.5,
            "height": "100%",
            "width": "100%",
          },
        ]
      }
    />
  </View>
</View>
`;
